src/OSPSuite.Presentation/Presenters/Charts/ChartEditorPresenter.cs

```diff
@@ -636,11 +644,15 @@ public void Handle(ChartUpdatedEvent chartUpdatedEvent)
          if (!canHandle(chartUpdatedEvent))
             return;
 
-         _curveColorGroupingPresenter.SetMetadata(getCommonMetaDataOfCurves());
```

Comment:
this should be called in the refresh I believe. Why do it BEFORE refreshing all the other presenters with data. It should be done last.. Since updateUsedColumns is called last, it should be good

Reply:
thanks for the changes @msevestre ! I totally agree with them. Sorry for taking so long to respond, I did not notice your comments at all - I thought this PR was over and did not have it in  mind. I will now merge with squash.